Library/Homebrew/formula.rb

```diff
@@ -494,7 +494,16 @@ def python(options={}, &block)
 
   # an array of all {Formula} names
   def self.names
-    Dir["#{HOMEBREW_LIBRARY}/Formula/*.rb"].map{ |f| File.basename f, '.rb' }.sort
+    core = Dir["#{HOMEBREW_LIBRARY}/Formula/*.rb"].map{ |f| File.basename f, '.rb' }.sort
+    taps = Dir["#{HOMEBREW_LIBRARY}/Taps/*/*/"].reduce([]) do |formulae, tap|
+      Pathname.new(tap).find_formula do |f|
+        if f.to_s =~ HOMEBREW_TAP_PATH_REGEX
+           formulae << "#$1/#{$2.gsub(/^homebrew-/, "")}/#{f.basename(".rb")}"
+        end
+      end
+      formulae
+    end.sort
+    core+taps
   end
```

Comment:
Since we drop the symlink, it also means formula whose name conflicts with core can be allowed. Therefore, using fully-qualified name to present tap formulae is a nature way and better way to do. Otherwise, it would be strange to return some of tapped formulae (that is we cannot predict which formula will return if some of them have the same name). On the other hand, passing fully-qualified name to caller can avoiding further path searching. It's good in term of performance.

For external commands, AFAIK, `homebrew-anytap` is broken (this is obvious) and `homebrew-livechek` failed to handle that path can be nil. I will report the issues to these projects after we are ready.

If we will make sure we don't break any external commands. How about only returning core formulae? Like this:
```ruby
def self.names options={:includetap => false}
   ...
  return core unless options[:includetap]
   ...
end
```